BonusCalcApi.Tests/IntegrationTests.cs

```diff
@@ -18,13 +19,19 @@ public abstract class IntegrationTests<TStartup> where TStartup : class
 
         private MockWebApplicationFactory<TStartup> _factory;
         private IDbContextTransaction _transaction;
-        private DbContextOptionsBuilder _builder;
+        private DbContextOptionsBuilder<BonusCalcContext> _builder;
 
         [OneTimeSetUp]
         public void OneTimeSetUp()
         {
-            _builder = new DbContextOptionsBuilder();
-            _builder.UseNpgsql(ConnectionString.TestDatabase())
+            _builder = new DbContextOptionsBuilder<BonusCalcContext>();
+
+            var dataSourceBuilder = new NpgsqlDataSourceBuilder(ConnectionString.TestDatabase());
+            dataSourceBuilder.MapEnum<BandChangeDecision>();
+            var dataSource = dataSourceBuilder.Build();
+
+            _builder
+                .UseNpgsql(dataSource)
```

Comment:
Your tests run may have different configuration than main app startup.
Startup has the: `UseQuerySplittingBehavior`, while this doesn't.
Depending on defaults this may or may not lead tests run to behave differently than the actual app would.